lib/search.js

```diff
@@ -53,8 +54,13 @@ function doSearch(query) {
         query.apps = query.apps.concat(apps);
         return query;
     }
-
-    return request(buildUrl(query))
+    var delay = h.calcDelay(optsGlobal.throttle, request._last);
+    return Promise
```

Comment:
Isn't there a way to avoid defining a promise twice? Plus it looks like you are delaying before making the first request. Maybe the delaying should be handled in the checkFinished step, before making the next call to doSearch